### ISSUE TITLE
Allow "$schema" via schema.json

### ISSUE THREAD

**Issue by GioCirque** [OAI/OpenAPI-Specification#1719]
The schema should allow the "$schema" element in the referencing document.

### Sample Document
```json
{
  "$schema": "https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/schemas/v2.0/schema.json",
  "swagger": "2.0",
  "info": {
    "title": "My Service API",
    "version": "0.0.1"
  },
  "paths": {}
}
```

### "Property $schema is not allowed."
```json
{
	"resource": "/Users/gpalacino/Source/play/service/service-api/openapi.json",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": null,
	"severity": 4,
	"message": "Property $schema is not allowed.",
	"startLineNumber": 2,
	"startColumn": 3,
	"endLineNumber": 2,
	"endColumn": 12
}
```

**Comment by MikeRalphson** [comment]
Can you elaborate on why you think this would be beneficial?

**Comment by handrews** [comment]
`$schema` is a JSON Schema keyword, and OAS is not, itself a JSON Schema (it is a document that contains JSON Schemas within it).  While some people do use a `$schema` property in JSON documents the same way that JSON Schema uses it, this is not actually any sort of standard and is, IMHO, more confusing than helpful.

**Comment by dubnemo** [comment]
Most people working with JSON Schema use it to explicitly call out which version of IEFT JSON Schema is being leveraged.  
  "$schema" : "http://json-schema.org/draft-04/schema#"
which always the meta-schema to validate the JSON Schema, and subsequent payload instances.  Very powerful stuff.

**Comment by MikeRalphson** [comment]
As above, `$schema` is used by JSON schema instances to identify the meta-schema, but it is not intended / standardised for use in JSON instance documents, and as the JSON schema for OpenAPI is informative only, linking to it from an OAS instance document would cause more problems than it solves.

**Comment by GioCirque** [comment]
There are a lot of valid points here and good information. I'd love to be able to reference the schema for intellisense and other editor conveniences, and I understand that I may be enjoying a side effect that shouldn't be explicitly supported.

**Comment by handrews** [comment]
@GioCirque @dubnemo a few things have happened to make this worth checking back in on:

* [JSON Schema 2019-09](http://json-schema.org/specification.html) (formerly known as "draft-08", but the "draft-NN" numbering [got confusing](http://json-schema.org/specification-links.html)) has been published.
* #1977 proposes making 2019-09 the basis for OpenAPI Specification 3.1's Schema Object, with the goal of full compatibility (in the sense that OpenAPI's Schema Object is a superset, which can be described using 2019-09's `$vocabulary` extensibility feature)
* In that sense, in my _personal_ opinion as JSON Schema spec editor, `$schema` should be allowed as long as the meta-schema named does not break compatibility with 2019-09 plus OAS 3.1's extensions.  The `$vocabulary` feature allows for verifying this.

In short, any meta-schema referenced in `$schema` within OAS 3.1 MUST declare vocabularies from the standard 2019-09 set, plus the OAS 3.1 extension vocabulary, optionally plus further extension vocabularies that conform to OAS's extension rules (e.g. `x-` keywords) and do not contradict the semantics of the 2019-09 + OAS 3.1 keywords.

@philsturgeon has there been any discussion of `$schema` with #1977?  I see that it's included but without any details.
